src/data_classifier.py

```diff
@@ -0,0 +1,173 @@
+        # check if it is within a close distance to a standard star
+        # if so, fix the object name to be the good one from the list!
+        try:
+            near_std, std_dist = wifes_calib.find_nearest_stdstar(fn)
+            if std_dist < 100.0:
+                obj_name = near_std
+        except:
+            pass
+        #---------------------------
+        # 1 - bias frames
+        if imagetype == 'BIAS':
+            bias.append(ob)
+        # 2 - quartz flats
+        if imagetype == 'FLAT':
+            domeflat.append(ob)
+        # 3 - twilight flats
+        if imagetype == 'SKYFLAT':
+            twiflat.append(ob)
+        # 4 - dark frames
+        if imagetype == 'DARK':
+            dark.append(ob)
+        # 5 - arc frames
+        if imagetype == 'ARC':
+            arc.append(ob)
+        # 6 - wire frames
+        if imagetype == 'WIRE':
+            wire.append(ob)
+        # 7 - standard star
+        if imagetype == 'STANDARD':
+            # group standard obs together!
+            if obj_name in stdstar.keys():
+                stdstar[obj_name].append(ob)
+            else:
+                stdstar[obj_name] = [ob]
+        
+        # all else are science targets (also consider standar star in imagety = OBJECT)
+        if imagetype == 'OBJECT':
+            if obj_name in stdstar_list:
+                # group standard obs together!
+                if obj_name in stdstar.keys():
+                    stdstar[obj_name].append(ob)
+                else:
+                    stdstar[obj_name] = [ob]
+            else:
+                # group science obs together!
+                if obj_name in science.keys():
+                    science[obj_name].append(ob)
+                else:
+                    science[obj_name] = [ob]
+
+    # #------------------
+    # science dictionay
+
+    sci_obs = []
+
+    for obj_name in science.keys():
+        obs_list = science[obj_name]
+        sci_obs.append({'sci':obs_list, 'sky':[]})
+
+
+    #------------------
+    # stdstars dictionary
+    std_obs = []
+
+    for obj_name in stdstar.keys():
+        obs_list = stdstar[obj_name]
+        std_obs.append({'sci':obs_list, 'name':obj_name,'type':['flux', 'telluric']})
+
+
+    obs_metadata = {
+        'bias' : bias,
+        'domeflat' : domeflat,
+        'twiflat' : twiflat,
+        'dark' : dark,
+        'wire' : wire,
+        'arc'  : arc,
+        'sci'  : sci_obs,
+        'std'  : std_obs}
+    
+    return obs_metadata 
+
+
+
+
+
+
+def classify(data_dir):
+
+    # What is this option for? TODO change to a default variable in the function
+    try:
+        naxis2_use = int(sys.argv[2])
+    except:
+        naxis2_use = 0
+
+    # Get list of all fits files in directory
+    all_files = os.listdir(data_dir)
+
+    # Filtering the data as per blue and red arm
+    blue_obs = []
+    red_obs = []
+
+    obs_date = None
+    for fn in all_files:
+        obs = fn.replace('.fits', '')
```

Comment:
This might be a change I could make quickly after the PR if you prefer. Quite a few renames and I'm not 100% on them.

Reply:
No worries, Felipe will do it tomorrow as he's the one who made the functions. Originally, he followed the same criteria for the variable names as in the pipeline, but yours are easier to read/understand. I only updated the first 2 suggestions, `all_files` and `fn`.

Reply:
All done here and the tests run ok. Thank you! We faked your changes in `src/wifes_calib.py` to avoid the `NaN` error in `extract_wifes_stdstar` and all ran ok.